components/announcement_bar/announcement_bar.jsx

```diff
@@ -239,10 +224,13 @@ export default class AnnouncementBar extends React.PureComponent {
             return;
         }
 
-        if (!prevState.allowDismissal && this.state.allowDismissal) {
-            document.body.classList.remove('announcement-bar--fixed');
-        } else if (prevState.allowDismissal && !this.state.allowDismissal) {
+        const wasFixed = this.shouldRender(prevProps, prevState) && !prevState.allowDismissal;
+        const isFixed = this.shouldRender(this.props, this.state) && !this.state.allowDismissal;
+
+        if (!wasFixed && isFixed) {
             document.body.classList.add('announcement-bar--fixed');
+        } else if (wasFixed && !isFixed) {
+            document.body.classList.remove('announcement-bar--fixed');
```

Comment:
No action required and more of a self-comment since I probably originally wrote this code, I wonder if it would be better in this instance to check `isFixed` and then check if the class exists on the body as the condition instead of comparing against previous state

Reply:
I'm guessing the root of the concern here is a potential mismatch between the DOM and the state of this component? That can definitely happen above given that we return early on `!this.props.isLoggedIn`, though upon reflection, I don't know why we do that. If we removed it, I think the lifecycle methods would guarantee the DOM would stay in sync.

My concern with using the DOM as the source of truth here would be related to performance and the inherent "anti-pattern" as such. Would you be amenable to keeping the logic "react side" and tweaking `componentDidUpdate` to always update the class here vs. skipping if logged out?